### PR TITLE
168 check idempotence on dappaddress

### PR DESCRIPTION
**Pull request by vscatarino**

### CODE REVIEW

File: internal/convenience/repository/input_repository_test.go [hunk 1]
```diff
@@ -309,6 +309,39 @@ func (s *InputRepositorySuite) TestCreateInputAndCheckAppContract() {
 	s.Equal("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266", input2.AppContract.Hex())
 }
 
+func (s *InputRepositorySuite) TestFindInputByAppContractAndIndex() {
+	defer s.teardown()
```
Comment (sandhilt) [inline internal/convenience/repository/input_repository_test.go:313]: Use Teardown from testify: https://pkg.go.dev/github.com/stretchr/testify/suite#TearDownTestSuite like SetupTest

**Review by sandhilt — APPROVED** [review]
LGTM 👍🏽

**Comment by fabiooshiro** [comment]
I think you're missing some other repositories

**Comment by vscatarino** [comment]
@fabiooshiro , it's about the app_contract, and this field exists only in the convenience_input table. Therefore, I created this just to generate the input in the convenience_table. I noticed that for other repositories (report and output), there are respective create functions using the idempotence approach.

**Pull request closed without merging**